src/VsixTesting.Xunit/Internal/VsTestCaseFactory.cs

```diff
@@ -80,18 +85,26 @@ internal static IEnumerable<Instance> GetInstances(IEnumerable<VsInstallation> i
 
         internal static IEnumerable<VsInstallation> FilterInstallations(IEnumerable<VsInstallation> installations, VsTestSettings settings, StringBuilder output = null)
         {
+            var parentVsInstallation = ParentVsProcess.Value == null ? null
+                : installations.FirstOrDefault(i => i.ApplicationPath.Equals(ParentVsProcess.Value.MainModule.FileName, StringComparison.OrdinalIgnoreCase));
+
             foreach (var group in installations.GroupBy(i => i.Version.Major).OrderBy(g => g.Key))
             {
-                var installation = group.OrderBy(i => i.Name.Contains("-pre")).First();
-
-                if (!settings.SupportedVersionRanges.Any(range => installation.Version >= range.Minimum && installation.Version <= range.Maximum))
+                foreach (var installation in group.OrderBy(i => i.Name.Contains("-pre")))
                 {
-                    output?.AppendLine($"Skipping {installation.Path} because the version {installation.Version} is not within any specified version range {string.Join(";", settings.SupportedVersionRanges)}.");
-                    continue;
+                    if (!IsVersionWithinRange(installation))
+                    {
+                        output?.AppendLine($"Skipping {installation.Path} because the version {installation.Version} is not within any specified version range {string.Join(";", settings.SupportedVersionRanges)}.");
+                        continue;
+                    }
+
+                    yield return parentVsInstallation != null && IsVersionWithinRange(parentVsInstallation) ? parentVsInstallation : installation;
```

Comment:
Won't this mean that if `parentVsInstallation` is within the target range, this same version will be returned multiple times? I noticed that is was no longer testing with 2015 when this code was in place.

Reply:
It will be returned once only bcz there is a `break` statement  after `yield return` but all this doesnt matter cuz ur implementation looks better :smile:

Reply:
I had indeed missed the `break` statement. 😉  What about the outer `foreach` statement though, won't it still return the same `parentVsInstallation` for each major version number? I assumed that was why it wasn't launching 2015 on my machine.

Reply:
I think so. I didn't happen in my first implementation cuz I was ensuring `installation.Version.Major == parentInstallation.Version.Major` but I swapped it for `IsVersionWithinRange` but I actually needed to keep both checks :smile:

Reply:
Added unit tests so that we can better sleep at night :smile:

Reply:
You're usually two steps ahead of me, so I'm glad I managed to catch this one! 😉